plugins/dotplot-view/src/DotplotView/blockTypes.ts

```diff
@@ -1,127 +0,0 @@
-import { sum } from '@jbrowse/core/util'
-
-type Func<T> = (value: BaseBlock, index: number, array: BaseBlock[]) => T
```

Comment:
this file fully deleted as it almost 100% matched packages/core/util/blockTypes.ts and was just slowly growing out of sync with it. the dotplot block types dont have any special behavior really

Reply:
full diff for posterity

```
1c1
< import { sum } from '@jbrowse/core/util'
---
> import { sum } from '.'
23a24,27
>   getRegions() {
>     return this.blocks.map(block => block.toRegion())
>   } ## extra method, in core
> 
39c43
<     return this.blocks.length
---
>     return this.blocks.length > 0  ## safety check, in core
43a48,57
>   get totalWidthPxWithoutBorders() {
>     return this.blocks.length > 0
>       ? sum(
>           this.blocks
>             .filter(block => block.variant !== 'boundary')
>             .map(blocks => blocks.widthPx),
>         )
>       : 0
>   }  ## extra method, in core
> 
45c59
<     return this.blocks.length ? this.blocks[0].offsetPx : 0
---
>     return this.blocks.length > 0 ? this.blocks[0].offsetPx : 0  ## safety check, in core
50a65,68
> 
>   get totalBp() {
>     return sum(this.contentBlocks.map(block => block.end - block.start))
>   } ## extra method, in core
53a72,73
>   public regionNumber?: number ## extra optional field, in core
> 
65a86,87
>   public offsetPx: number ## offsetPx not initialized by default to 0 in core version here, but looks inconsequential
> 
68c90,92
<   public offsetPx = 0
---
>   public variant?: string
> 
>   public isLeftEndOfDisplayedRegion?: boolean
80a105
>     this.offsetPx = data.offsetPx

```